tests/test_config/test_partial_config.py

```diff
@@ -0,0 +1,118 @@
+import json
+import tempfile
+
+import pytest
+from libsonata._libsonata import SonataError
+
+import bluepysnap.circuit as test_module
+from bluepysnap.exceptions import BluepySnapError
+
+from utils import copy_config, edit_config, copy_test_data
+
+
+def test_partial_circuit_config_wrong_status():
+    with copy_config("circuit_config.json") as config_path:
+        with edit_config(config_path) as config:
+            config["metadata"] = {"status": "NOT A TYPE"}
+
+        with pytest.raises(SonataError):
+            test_module.Circuit(config_path)
+
+
+@pytest.mark.parametrize(
+    "config",
+    [
+        {"metadata": {"status": "partial"}},
+        {
+            "metadata": {"status": "partial"},
+            "components": {
+                "morphologies_dir": "some/morph/dir",
+            },
+            "networks": {
+                "nodes": [],
+            },
+        },
+        {
+            "metadata": {"status": "partial"},
+            "components": {
+                "morphologies_dir": "some/morph/dir",
+            },
+            "networks": {
+                "edges": [],
+            },
+        },
+    ],
+)
+def test_partial_circuit_config_empty(config):
```

Comment:
If nodes/edges listing is missing or it is empty, empty lists or errors are returned (I have added one). Unless we want to disable Circuit running empty entirely, I think it can stays as is.